src/main/java/ru/sbt/mipt/oop/Application.java

```diff
@@ -11,58 +11,74 @@
 public class Application {
 
     public static void main(String... args) throws IOException {
-        // считываем состояние дома из файла
-        Gson gson = new Gson();
-        String json = new String(Files.readAllBytes(Paths.get("smart-home-1.js")));
-        SmartHome smartHome = gson.fromJson(json, SmartHome.class);
+        SmartHome smartHome = readSmartHomeFromFile();
+
         // начинаем цикл обработки событий
         SensorEvent event = getNextSensorEvent();
         while (event != null) {
             System.out.println("Got event: " + event);
             if (event.getType() == LIGHT_ON || event.getType() == LIGHT_OFF) {
-                // событие от источника света
-                for (Room room : smartHome.getRooms()) {
-                    for (Light light : room.getLights()) {
-                        if (light.getId().equals(event.getObjectId())) {
-                            if (event.getType() == LIGHT_ON) {
-                                light.setOn(true);
-                                System.out.println("Light " + light.getId() + " in room " + room.getName() + " was turned on.");
-                            } else {
-                                light.setOn(false);
-                                System.out.println("Light " + light.getId() + " in room " + room.getName() + " was turned off.");
-                            }
-                        }
+                doLightsAction(smartHome, event);
+            } else {
+                doDoorActions(smartHome, event);
+            }
+            event = getNextSensorEvent();
+        }
+    }
+
+    private static SmartHome readSmartHomeFromFile() throws IOException {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D0%BD%D0%B5-%D0%B8%D1%81%D0%BF%D0%BE%D0%BB%D1%8C%D0%B7%D1%83%D0%B9%D1%82%D0%B5-%D1%81%D1%82%D0%B0%D1%82%D0%B8%D1%87%D0%B5%D1%81%D0%BA%D0%B8%D0%B5-%D0%BC%D0%B5%D1%82%D0%BE%D0%B4%D1%8B-%D0%B2-%D0%B1%D0%B8%D0%B7%D0%BD%D0%B5%D1%81-%D0%BB%D0%BE%D0%B3%D0%B8%D0%BA%D0%B5